src/bika/lims/browser/fields/resultrangefield.py

```diff
@@ -96,16 +97,25 @@ def __call__(self):
         # Get the AnalysisRequest to look at
         analysis = self.context
         sample = analysis.getRequest()
-        if not sample:
+        results_range = self.get_results_range(sample)
+        if not results_range:
             return {}
 
         # Search by keyword
-        field = sample.getField("ResultsRange")
         keyword = analysis.getKeyword()
-        rr = field.get(sample, search_by=keyword)
+        rr = results_range.get(sample, search_by=keyword)
         if rr:
             return rr
 
         # Try with uid (this shouldn't be necessary)
         service_uid = analysis.getServiceUID()
-        return field.get(sample, search_by=service_uid) or {}
+        return results_range.get(sample, search_by=service_uid) or {}
+
+    def get_results_range(self, sample):
+        results_range = None
+        try:
+            results_range = sample.ResultsRange
```

Comment:
Please always use the field getter instead of accessing the attribute directly.

Maybe you can simplify the whole thing with a combination this:

```
 def get_results_range_for(self, sample, analysis):
    rr = {}
    try:
        field = sample.getField("ResultsRange", {})
        rr = field.get(sample, search_by=analysis.getKeyword())
     except AttributeError:
        pass
     return rr
```

Also use a more descriptive logging message. As you know that it is a sample, you could be more specific, e.g.

```
logger.error("Failed to get results range for sample '{}' and analysis '{}': "
             "{}".format(sample.getId(), analysis.getKeyword(), str(e)))
```

Reply:
So, if at look on this code:
```python
# Try with uid (this shouldn't be necessary)
service_uid = analysis.getServiceUID()
```

, getting the results range is solved by checking the existence of the `Keyword`.  Right?

Or to get results range by UID has the meaning?

Reply:
Good question. I'm not sure why we added this fallback. But maybe you can bake it in for backwards compatibility?